### PR TITLE
convert structs with single operator() to funcs

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: src/download/download_constructor.cc [hunk 1]
```diff
@@ -57,28 +57,19 @@
...
 
-struct download_constructor_is_single_path {
-  bool operator () (Object::map_type::const_reference v) const {
-    return
-      std::strncmp(v.first.c_str(), "name.", sizeof("name.") - 1) == 0 &&
-      v.second.is_string();
-  }
+bool
+download_constructor_is_single_path(Object::map_type::const_reference v) {
+  return v.first.find("name.") == 0 && v.second.is_list();
```
Comment (rakshasa) [inline src/download/download_constructor.cc:62]: This should be `is_string`.

File: src/torrent/tracker_list.cc [hunk 2]
```diff
@@ -84,14 +84,14 @@ TrackerList::has_active_not_scrape_in_group(uint32_t group) const {
...
-  bool operator () (const TrackerList::value_type& value) const { return value->is_usable(); }
+bool
+tracker_usable_t(const TrackerList::value_type& value) {
+  return value->is_usable();
 };
 
 bool
 TrackerList::has_usable() const {
-  return std::find_if(begin(), end(), tracker_usable_t()) != end();
+  return std::any_of(begin(), end(), tracker_usable_t);
```
Comment (rakshasa) [inline src/torrent/tracker_list.cc:94]: Simple functions like this should be using lambda functions.

File: src/download/download_constructor.cc (hunk 1 above)
Comment (neheb) [reply src/download/download_constructor.cc:62]: Huh. I wonder how that slipped in.

File: src/torrent/tracker_list.cc (hunk 2 above)
Comment (neheb) [reply src/torrent/tracker_list.cc:94]: Will do. I avoided doing so as to not change it too much.

File: src/download/download_main.cc [hunk 3]
```diff
@@ -424,11 +421,10 @@ DownloadMain::do_peer_exchange() {
     // Create the new m_ut_pex_list by removing any 'removed'
     // addresses from the original list and then adding the new
     // addresses.
-    m_ut_pex_list.erase(std::set_difference(m_ut_pex_list.begin(), m_ut_pex_list.end(), removed.begin(), removed.end(),
-                                            m_ut_pex_list.begin(), SocketAddressCompact_less()), m_ut_pex_list.end());
+    m_ut_pex_list.erase(std::set_difference(m_ut_pex_list.begin(), m_ut_pex_list.end(), removed.begin(), removed.end(), m_ut_pex_list.begin(), SocketAddressCompact_less), m_ut_pex_list.end());
```
Comment (rakshasa) [inline src/download/download_main.cc:424]: The `set_difference` calls are so long they should be multiline, with each begin/end and ftor on separate lines.

Comment (neheb) [reply src/download/download_main.cc]: This is the best I could get clang-format to do.

**Comment by neheb** [comment]
Switched to std::mem_fn instead of lambdas where shorter.

**Pull request merged**